examples/jsm/loaders/MMDLoader.js

```diff
@@ -2064,4 +2076,114 @@ class CubicBezierInterpolation extends Interpolant {
 
 }
 
+class MMDToonMaterial extends ShaderMaterial {
+
+	constructor( parameters ) {
+
+		super();
+
+		this._matcapCombine = AddOperation;
+		this.emissiveIntensity = 1.0;
+		this.normalMapType = TangentSpaceNormalMap;
+
+		this.combine = MultiplyOperation;
+
+		this.wireframeLinecap = 'round';
+		this.wireframeLinejoin = 'round';
+
+		this.flatShading = false;
+
+		this.lights = true;
+
+		this.vertexShader = MMDToonShader.vertexShader;
+		this.fragmentShader = MMDToonShader.fragmentShader;
+
+		this.defines = Object.assign( {}, MMDToonShader.defines );
+		Object.defineProperties( this, {
+
+			matcapCombine: {
+
+				get: function () {
+
+					return this._matcapCombine;
+
+				},
+
+				set: function ( value ) {
+
+					this._matcapCombine = value;
+
+					switch ( value ) {
+
+						case MultiplyOperation:
+							this.defines.MATCAP_BLENDING_MULTIPLY = true;
+							delete this.defines.MATCAP_BLENDING_ADD;
+							break;
+
+						default:
+						case AddOperation:
+							this.defines.MATCAP_BLENDING_ADD = true;
+							delete this.defines.MATCAP_BLENDING_MULTIPLY;
+							break;
+
+					}
+
+				}
+
+			},
+
+		} );
+
+		this.uniforms = UniformsUtils.clone( MMDToonShader.uniforms );
+		for ( const uniformName in this.uniforms ) {
+
+			Object.defineProperties( this, {
+
+				[ uniformName ]: {
+
```

Comment:
I think it would be simpler to use `Object.defineProperty`

```javascript
Object.defineProperty( this, uniformName, {
   get: ...,
   set: ...
} );
```

Reply:
hmm..., I got this error when I try this:
```js
Uncaught TypeError: Property description must be an object: d
    at Function.defineProperties (<anonymous>)
    at new MMDToonMaterial (MMDLoader.js?aed8:2142)
    at MaterialBuilder.build (MMDLoader.js?aed8:1278)
    at MeshBuilder.build (MMDLoader.js?aed8:423)
    at eval (MMDLoader.js?aed8:146)
    at Object.eval [as onLoad] (MMDLoader.js?aed8:254)
    at XMLHttpRequest.eval (three.module.js?5a89:35084)
```